python_modules/dagster/docs/reference.rst

```diff
@@ -0,0 +1,212 @@
+Reference
+---------
+
+
+Solid
+^^^^^
+
+A solid is a functional unit of computation. Solids define their (optionally typed) inputs
+and outputs and the typed schema by which they can be configured, and can enfore expectations on
+their outputs.
+
+Solids can be strung together into `pipelines <#pipeline>`__ by defining
+`dependencies <#dependency-definition>`__ between their inputs and outputs.  Solids are reusable
+and instances of a solid may appear many times in a given pipeline, or across many different
+pipelines.
+
+Solids often wrap code written in or intended to execute in other systems (e.g., SQL statements,
+Jupyter notebooks, or Spark jobs written in Scala), providing a common interface for defining,
+orchestrating, and managing data processing applications with heterogeneous components
+
+Solids are defined using the :func:`@lambda_solid <dagster.lambda_solid>` or
+:func:`@solid <dagster.solid>` decorators, or using the underlying
+:class:`SolidDefinition <dagster.SolidDefinition>` class. These APIs wrap an underlying
+`transform function`, making its metadata queryable by higher-level tools.
+
+Transform Function
+^^^^^^^^^^^^^^^^^^
+
+The user-supplied function which forms the heart of a solid definition and will be executed when
+the solid is invoked by the Dagster engine.
+
+
+Result
+^^^^^^
+
```

Comment:
Maybe start with what a Result is before the rules of how its used "Result is how a Solid's transform function communicates results to dagster. It contains the result value and an output name." or something

Reply:
yep